roles/static_inventory/tasks/openstack.yml

```diff
@@ -40,8 +44,25 @@
       add_host:
         name: '{{ item.name }}'
         groups: '{{ item.metadata.group }}'
-        ansible_host: '{{ item.public_v4 }}'
+        ansible_host: "{% if use_bastion|bool %}{{ item.name }}{% else %}{{ item.private_v4 }}{% endif %}"
         ansible_fqdn: '{{ item.name }}'
+        ansible_user: '{{ ssh_user }}'
         ansible_private_key_file: '{{ private_ssh_key }}'
+        ansible_ssh_extra_args: '-F {{ ssh_config_path }}'
```

Comment:
Right, we can't omit it for all servers with floating IPs because infra and masters still need a floating IP even in the bastion scenario.

So yeah, the ssh_extra_args should be set only when we have bastion enabled.